spec/models/bundle_context_spec.rb

```diff
@@ -17,64 +17,44 @@
       expect(BundleContext.new).not_to be_valid
       expect(bc).to be_valid
     end
-
-    context "defines enum with expected values" do
-      it "content_structure enum" do
-        is_expected.to define_enum_for(:content_structure).with(
-          "simple_image" => 0,
-          "simple_book" => 1,
-          "book_as_image" => 2,
-          "file" => 3,
-          "smpl" => 4
-        )
-      end
-
-      it "content_metadata_creation enum" do
-        is_expected.to define_enum_for(:content_metadata_creation).with(
-          "default" => 0,
-          "filename" => 1,
-          "smpl_cm_style" => 2
-        )
-      end
+    it 'is not valid without a User' do
+      expect { bc.user = nil }.to change(bc, :valid?).to(false)
     end
-
-    describe "#content_structure=" do
-      it "validation rejects a value if it does not match the enum" do
-        expect { described_class.new(content_structure: 654) }
-          .to raise_error(ArgumentError, "'654' is not a valid content_structure")
-        expect { described_class.new(content_structure: 'book_as_pdf') }
-          .to raise_error(ArgumentError, "'book_as_pdf' is not a valid content_structure")
-      end
-
-      it "will accept a symbol, but will always return a string" do
-        expect(described_class.new(content_structure: :smpl).content_structure).to eq 'smpl'
-      end
+    it 'is not valid unless bundle_dir exists on filesystem' do
+      expect { bc.bundle_dir = 'does/not/exist' }.to change(bc, :valid?).to(false)
     end
+  end
 
-    describe "#content_metadata_creation=" do
-      it "validation rejects a value if it does not match the enum" do
-        expect { described_class.new(content_metadata_creation: 654) }
-          .to raise_error(ArgumentError, "'654' is not a valid content_metadata_creation")
-        expect { described_class.new(content_metadata_creation: 'dpg') }
-          .to raise_error(ArgumentError, "'dpg' is not a valid content_metadata_creation")
-      end
-
-      it "will accept a symbol, but will always return a string" do
-        expect(described_class.new(content_metadata_creation: :smpl_cm_style).content_metadata_creation).to eq 'smpl_cm_style'
-      end
-    end
+  it do
+    is_expected.to define_enum_for(:content_structure).with(
+      "simple_image" => 0,
+      "simple_book" => 1,
+      "book_as_image" => 2,
+      "file" => 3,
+      "smpl" => 4
+    )
+  end
+  it do
+    is_expected.to define_enum_for(:content_metadata_creation).with(
+      "default" => 0,
+      "filename" => 1,
+      "smpl_cm_style" => 2
+    )
+  end
 
-    context "bundle_dir path does not exist" do
-      it "object does not pass validation" do
-        expect { bc.bundle_dir = 'does/not/exist' }.to change { bc.valid? }.to(false)
-      end
-    end
```

Comment:
since the bundle_dir path existing on filesystem validation was a bespoke method, this test should stay

Reply:
Test is still present in line 23 in the (now much smaller) `validation` section.

Reply:
I should also say I think that checking the filesystem is a problematic thing to include specifically in model validation.  It means that the same record, same data is valid or invalid... depending on some un-modeled outside thing.  Previously `valid` data was saved, but now could be invalid.  But the whole point of validations guarantee **only** valid data is in the database.  If the validation cannot do that (because it is falsifiable by outside conditions), then it isn't really an ORM model validation.  

This is an important point, so I hope I have described it adequately.  Just because AR validation machinery is nearby doesn't make it a fit for this particular check.

Alternatives: check in `before_save`, check in controller `create`, etc.

Reply:
filed #307 as a backlog ticket